app/src/main/java/com/kickstarter/ui/activities/SignupActivity.kt

```diff
@@ -0,0 +1,128 @@
+
+        viewModel.outputs.formSubmitting()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { setFormDisabled(it) }
+
+        viewModel.outputs.formIsValid()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { setFormEnabled(it) }
+
+        viewModel.outputs.sendNewslettersIsChecked()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                SwitchCompatUtils.setCheckedWithoutAnimation(binding.signupFormView.newsletterSwitch, it)
+            }
+
+        viewModel.outputs.errorString()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { ViewUtils.showDialog(this, null, it) }
+
+        RxView.clicks(binding.signupFormView.newsletterSwitch)
+            .skip(1)
+            .compose(bindToLifecycle())
+            .subscribe {
+                viewModel
+                    .inputs
+                    .sendNewslettersClick(binding.signupFormView.newsletterSwitch.isChecked)
+            }
+
+        binding.signupFormView.disclaimer.setOnClickListener {
+            disclaimerClick()
+        }
+
+        binding.signupFormView.signupButton.setOnClickListener {
+            signupButtonOnClick()
+        }
+
+        binding.signupFormView.name.doOnTextChanged { name, _, _, _ ->
+            name?.let { onNameTextChanged(it) }
+        }
+
+        binding.signupFormView.email.doOnTextChanged { email, _, _, _ ->
+            email?.let { onEmailTextChanged(it) }
+        }
+
+        binding.signupFormView.password.doOnTextChanged { password, _, _, _ ->
+            password?.let { onPasswordTextChange(it) }
+        }
+    }
+
+    private fun disclaimerClick() {
+        LoginPopupMenu(this, binding.loginToolbar.helpButton).show()
+    }
+
+    private fun onNameTextChanged(name: CharSequence) {
+        viewModel.inputs.name(name.toString())
+    }
+
+    private fun onEmailTextChanged(email: CharSequence) {
+        viewModel.inputs.email(email.toString())
+    }
+
+    private fun onPasswordTextChange(password: CharSequence) {
+        viewModel.inputs.password(password.toString())
+    }
+
+    private fun signupButtonOnClick() {
+        viewModel.inputs.signupClick()
+        this.hideKeyboard()
+    }
+
+    private fun onSuccess() {
+        setResult(RESULT_OK)
+        finish()
+    }
+
+    private fun setFormEnabled(enabled: Boolean) {
+        binding.signupFormView.signupButton.isEnabled = enabled
+    }
+
+    private fun setFormDisabled(disabled: Boolean) {
+        setFormEnabled(!disabled)
+    }
+
+    override fun exitTransition(): Pair<Int, Int>? {
+        return TransitionUtils.slideInFromLeft()
+    }
+
+    override fun back() {
+        if (this.supportFragmentManager.backStackEntryCount == 0) {
```

Comment:
this if condition can be deleted, it was mean to not allow pressing back button on the `Interstitial Fragment` that screen is no longer presented